docs/metrics-api-design.md

```diff
@@ -0,0 +1,519 @@
+   */
+  nostd::shared_ptr<Meter> GetMeter(nostd::string_view library_name,
+                                    nostd::string_view library_version = "")
+};
+```
+
+Using this MeterProvider, users can obtain new Meters through the GetMeter function.
+
+## **Metric Instruments (`Meter` Class)**
+
+**Metric Events**
+
+This interface consists of a set of **instrument constructors**, and a **facility for capturing batches of measurements.**
+
+
+```
+# meter.h
+
+Class Meter {
+public:
+  
+/////////////////////////Metric Instrument Constructors////////////////////////////  
+  
+ /*
+  * New Double Counter
+  *
+  * Function that creates and returns a Counter metric instrument with value
+  * type double.
+  *
+  * Arguments:
+  * name, the name of the metric instrument (must conform to the above syntax).
+  * description, a brief, readable description of the metric instrument.
+  * unit, the unit of metric values following the UCUM convention
+  *       (https://unitsofmeasure.org/ucum.html).
+  * enabled, a boolean that turns on or off collection.
+  *
+  */ 
+  virtual nostd::unique_ptr<DoubleCounter> 
+                            NewDoubleCounter(nostd::string_view name, 
+                            nostd::string_view description,
+                            nostd::string_view unit, 
+                            nostd::string_view enabled);
+  
+ /*
+  * New int Counter
+  *
+  * Function that creates and returns a Counter metric instrument with value
+  * type int.
+  *
+  * Arguments:
+  * name, the name of the metric instrument (must conform to the above syntax).
+  * description, a brief, readable description of the metric instrument.
+  * unit, the unit of metric values following the UCUM convention
+  *       (https://unitsofmeasure.org/ucum.html).
+  * enabled, a boolean that turns on or off collection.
+  *
+  */ 
+  virtual nostd::unique_ptr<LongCounter> NewIntCounter(nostd::string_view name,
+                                                       nostd::string_view description,
+                                                       nostd::string_view unit,
+                                                       nostd::string_view enabled);
+  
+////////////////////////////////////////////////////////////////////////////////////
+//                                                                                //
+//                     Repeat above two functions for all                         //
+//                     six (five other) metric instruments.                       //
+//                                                                                //
+////////////////////////////////////////////////////////////////////////////////////
+  
+private:
+
+ /*
+  * CollectAll
+  *
+  * Checkpoints all metric instruments created from this meter and returns a
+  * vector of records containing the name, labels, and values of each instrument.
+  *
+  */
+  virtual nostd::vector<Record> CollectAll() = 0;
```

Comment:
Do we need/want this in the public API interface?

Reply:
No, we do not. This was mistakenly left in. Will update.